src/Pages/AuthPages/Register/Register.tsx

```diff
@@ -1,3 +1,259 @@
+import {
+  AlternateEmail,
+  LanguageOutlined,
+  Person2Outlined,
+  PhoneInTalk,
+} from "@mui/icons-material";
+import { Box, Stack, Typography } from "@mui/material";
+import { useEffect } from "react";
+import { FileUploader } from "react-drag-drop-files";
+import { FieldValues, SubmitHandler, useForm } from "react-hook-form";
+import { Link } from "react-router-dom";
+import { AUTHENTICATION_URLS } from "../../../Api/END_POINTS.tsx";
+import ButtonForm from "../../../Components/SharedComponents/ButtonForm/ButtonForm.tsx";
+import { FormTextField } from "../../../Components/SharedComponents/FormTextField/FormTextField";
+import { PasswordTextField } from "../../../Components/SharedComponents/PasswordTextField/PasswordTextField.tsx";
+import { useFetch } from "../../../Context/FetchContext";
+
 export default function Register() {
-  return <div>Register</div>;
+  const fileTypes = ["JPG", "PNG", "GIF"];
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    setError,
+    clearErrors,
+    watch,
+    formState: { errors },
+    setFocus,
+  } = useForm();
+  useEffect(() => {
+    setFocus("userName");
+  }, [setFocus]);
+  const { fetchData, loading } = useFetch();
+
+  const onSubmit: SubmitHandler<FieldValues> = async (data) => {
+    const formData = new FormData();
+
+    if (data.profileImage === undefined) {
+      setError("profileImage", {
+        type: "manual",
+      });
+
+      return;
+    }
+    clearErrors("profileImage");
+    formData.append("userName", data.userName);
+    formData.append("email", data.email);
+    formData.append("country", data.country);
+    formData.append("phoneNumber", data.phoneNumber);
+    formData.append("password", data.password);
+    formData.append("confirmPassword", data.confirmPassword);
+    formData.append("profileImage", data.profileImage ? data.profileImage : "");
+    formData.append("role", "user");
+
+    fetchData({
+      method: "POST",
+      url: AUTHENTICATION_URLS.regitser,
+      showToastify: true,
+      data: formData,
+      navigateTo: "/auth/login",
+    });
+  };
+
+  const handleChange = (file: {
```

Comment:
Th File type is a built in type, we can make it `file:File`

Reply:
done